New fields, catch more cases, float support, and update dependencies

QuixThe2nd opened the pull request.

clement-escolano commented:
Thank you for your pull request! I have a few comments, happy to discuss

> I'd like to add support for multiple audio specs, but this would be a breaking change, changing audio from `string` to `string[]`. Is this something you'd be open to? @clement-escolano

I think it would be simpler to have `audio_list` or something similar to provide a simple interface in the general case and a more complex one for those who need it

QuixThe2nd commented:
Alright, I addressed all the feedback except `WEB`. Unsure of what to do there, Im not certain all release groups are consistent about their WEB naming.

QuixThe2nd commented:
Should be finished.

One more thing I wanted to run by you, I've actually been working on a TypeScript rewrite of this project on a [separate branch](https://github.com/QuixThe2nd/parse-torrent-title/tree/typescript). Since I'm planning to contribute a lot more features going forward, it would make development much easier (better tooling, type safety, etc.). Would you be interested in migrating to TypeScript?

clement-escolano commented:
Perfect thank you, I just published a new release with your changes.

I understand the reason to migrate to TypeScript but unsure I want to do it because I like the simplicity of the package as it is and it would add a compile step that would make it more complex to handle.
